ffi/capi/dart/package/lib/src/ICU4XAnyCalendarKind.dart

```diff
@@ -0,0 +1,164 @@
+import 'ICU4XError.dart';
+import 'ICU4XLocale.dart';
+import 'dart:convert';
+import 'dart:ffi' as ffi;
+import 'diplomat_runtime.dart';
+import 'package:ffi/ffi.dart' as allocators;
+
+typedef ICU4XAnyCalendarKindFfi = int;
+
+/// The various calendar types currently supported by [`ICU4XCalendar`]
+///
+/// See the [Rust documentation for `AnyCalendarKind`](https://docs.rs/icu/latest/icu/calendar/enum.AnyCalendarKind.html) for more information.
+enum ICU4XAnyCalendarKind {
+  /// The kind of an Iso calendar
+  Iso._(0),
+
+  /// The kind of a Gregorian calendar
+  Gregorian._(1),
+
+  /// The kind of a Buddhist calendar
+  Buddhist._(2),
+
+  /// The kind of a Japanese calendar with modern eras
+  Japanese._(3),
+
+  /// The kind of a Japanese calendar with modern and historic eras
+  JapaneseExtended._(4),
+
+  /// The kind of an Ethiopian calendar, with Amete Mihret era
+  Ethiopian._(5),
+
+  /// The kind of an Ethiopian calendar, with Amete Alem era
+  EthiopianAmeteAlem._(6),
+
+  /// The kind of a Indian calendar
+  Indian._(7),
+
+  /// The kind of a Coptic calendar
+  Coptic._(8),
+
+  /// The kind of a Dangi calendar
+  Dangi._(9),
+
+  /// The kind of a Chinese calendar
+  Chinese._(10),
+
+  /// The kind of a Hebrew calendar
+  Hebrew._(11),
+
+  /// The kind of a Islamic civil calendar
+  IslamicCivil._(12),
+
+  /// The kind of a Islamic observational calendar
+  IslamicObservational._(13),
+
+  /// The kind of a Islamic tabular calendar
+  IslamicTabular._(14),
+
+  /// The kind of a Islamic Umm al-Qura calendar
+  IslamicUmmAlQura._(15),
+
+  /// The kind of a Persian calendar
+  Persian._(16),
+
+  /// The kind of a Roc calendar
+  Roc._(17);
+
+  const ICU4XAnyCalendarKind._(this._id);
+
+  final int _id;
+
+  /// Read the calendar type off of the -u-ca- extension on a locale.
+  ///
+  /// Errors if there is no calendar on the locale or if the locale's calendar
+  /// is not known or supported.
+  ///
+  /// See the [Rust documentation for `get_for_locale`](https://docs.rs/icu/latest/icu/calendar/enum.AnyCalendarKind.html#method.get_for_locale) for more information.
+  factory ICU4XAnyCalendarKind.getForLocale(ICU4XLocale locale) {
+    final result = _getForLocaleFfi(ICU4XLocaleAsFfi(locale));
+    return result.isOk
+        ? ICU4XAnyCalendarKindFromFfi(result.union.ok)
+        : throw VoidError();
```

Comment:
Question: is throwing a VoidError the right behavior here? The function in Rust I think returns an Option

EDIT: Ok, it is currently a Result because we don't have Diplomat Option support fully fleshed out yet...

Reply:
I can treat `Result<T, ()>` as `Option<T>`, which in Dart would become `T?` (nullable)

Reply:
Filed https://github.com/rust-diplomat/diplomat/issues/354